### PR TITLE
Set tracing when DebugLevel > 0

### PR DESCRIPTION
**Pull request by caglar10ur**
Otherwise serial package creates so much noise.

### CODE REVIEW

File: pkg/serial/handshake.go [hunk 1]
```diff
@@ -40,6 +40,9 @@ const (
 
 // PurgeIncoming is used to clear a channel of bytes prior to handshaking
 func PurgeIncoming(conn net.Conn) {
+	if tracing {
```
Comment (fdawg4l) [inline pkg/serial/handshake.go]: BTW, you could modify `trace.newTrace`, make it public, and wrap this `if() { trace... }` in function with an appropriate skip (3 looks right).

**Comment by fdawg4l** [comment]
LGTM

File: pkg/serial/handshake.go (hunk 1 above)
Comment (caglar10ur) [inline pkg/serial/handshake.go]: But that would either require passing a new param to Begin, or doing more calls to trace hence breaking that defer End(Begin(msg)) chain

**Comment by anchal-agrawal** [comment]
:shipit:

**Comment by caglar10ur** [comment]
♻️

**Pull request merged**